CMake Error: Target X86 is not in the set of libraries

hi,
I tried compiling master on archlinux (x86_64, cmake 3.26, gcc 13, llvm 15) and the configuration step fails:
```
$ yay -Syu clang llvm zlib ncurses xeus-zmq cmake cppzmq
$ git clone --depth 1 https://github.com/lfortran/lfortran.git
$ cd lfortran
$ cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DWITH_LLVM=ON -DWITH_RUNTIME_LIBRARY=ON -DWITH_XEUS=ON -DWITH_ZLIB=ON .
CMake Error at CMakeLists.txt:11 (file):
  file STRINGS file "/tmp/lfortran/version" cannot be read.


-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /usr/lib/libz.a (found version "1.2.13")  
-- Performing Test HAVE_FFI_CALL
-- Performing Test HAVE_FFI_CALL - Success
-- Found FFI: /usr/lib/libffi.so  
-- Performing Test Terminfo_LINKABLE
-- Performing Test Terminfo_LINKABLE - Success
-- Found Terminfo: /usr/lib/libtinfo.so  
-- Found zstd: /usr/lib/libzstd.so  
-- Found LibXml2: /usr/lib/libxml2.so (found version "2.10.4") 
-- Found LLVM 15.0.7
-- Using LLVMConfig.cmake in: /usr/lib/cmake/llvm
CMake Error at /usr/lib/cmake/llvm/LLVM-Config.cmake:138 (message):
  Target X86 is not in the set of libraries.
Call Stack (most recent call first):
  /usr/lib/cmake/llvm/LLVM-Config.cmake:263 (llvm_expand_pseudo_components)
  CMakeLists.txt:172 (llvm_map_components_to_libnames)


-- Configuring incomplete, errors occurred!
```
any idea what happens ?

I also tried latest tag with the same result

yes, build0 succeeded, but now it's pretty much the same error:
```
$ cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_LLVM=yes -DCMAKE_INSTALL_PREFIX=`pwd`/inst .
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /usr/lib/libz.a (found version "1.2.13")  
-- Performing Test HAVE_FFI_CALL
-- Performing Test HAVE_FFI_CALL - Success
-- Found FFI: /usr/lib/libffi.so  
-- Performing Test Terminfo_LINKABLE
-- Performing Test Terminfo_LINKABLE - Success
-- Found Terminfo: /usr/lib/libtinfo.so  
-- Found zstd: /usr/lib/libzstd.so  
-- Found LibXml2: /usr/lib/libxml2.so (found version "2.10.4") 
-- Found LLVM 15.0.7
-- Using LLVMConfig.cmake in: /usr/lib/cmake/llvm
CMake Error at /usr/lib/cmake/llvm/LLVM-Config.cmake:138 (message):
  Target X86 is not in the set of libraries.
Call Stack (most recent call first):
  /usr/lib/cmake/llvm/LLVM-Config.cmake:263 (llvm_expand_pseudo_components)
  CMakeLists.txt:172 (llvm_map_components_to_libnames)


-- Configuring incomplete, errors occurred!
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,17 @@ if (WITH_LLVM)
         add_definitions("-DHAVE_TARGET_WASM=1")
     endif()
 
-    llvm_map_components_to_libnames(llvm_libs ${LFORTRAN_LLVM_COMPONENTS})
+    if (TARGET LLVMCore)
+        # If `LLVMCore` target is present, then LLVM is distributed as separate
+        # libraries and llvm_map_components_to_libnames() should work:
+        llvm_map_components_to_libnames(llvm_libs ${LFORTRAN_LLVM_COMPONENTS})
+    else()
+        # Workaround for https://github.com/llvm/llvm-project/issues/34593
+        # If LLVM is distributed as a single library (the LLVMCore target is
+        # missing), we set `llvm_libs` to "LLVM" which links against the single
+        # `libLLVM.so` shared library.
+        set(llvm_libs "LLVM")
+    endif()
     unset(LFORTRAN_LLVM_COMPONENTS)
 
     add_library(p::llvm INTERFACE IMPORTED)

